knowage-vue/src/modules/qbe/QBE.vue

```diff
@@ -0,0 +1,918 @@
+            this.mainQuery.subqueries.push(newSubquery)
+            this.selectedQuery = newSubquery
+            this.queryPreviewData = null as any
+        },
+        createQueryName() {
+            var lastcount = 0
+            var lastIndex = this.mainQuery.subqueries?.length - 1
+            if (lastIndex != -1) {
+                var lastQueryId = this.mainQuery.subqueries[lastIndex].id
+                lastcount = parseInt(lastQueryId.substr(1))
+            } else {
+                lastcount = 1
+            }
+            return lastcount + 1
+        },
+        async onExecute(qbeParameters: any[]) {
+            if (this.qbe) {
+                this.qbe.pars = [...qbeParameters]
+                await this.loadQBE()
+                this.qbeLoaded = true
+                this.parameterSidebarVisible = false
+            }
+        },
+        async openPreviewDialog() {
+            this.pagination.limit = 20
+            await this.executeQBEQuery()
+            this.qbePreviewDialogVisible = true
+        },
+        updateSmartView() {
+            this.smartView ? this.executeQBEQuery() : ''
+        },
+        smartViewFieldHidden() {
+            this.checkIfHiddenColumnsExist()
+            this.updateSmartView()
+        },
+        smartViewReorder(event) {
+            var temp = this.selectedQuery.fields[event.dragIndex]
+            this.selectedQuery.fields[event.dragIndex] = this.selectedQuery.fields[event.dropIndex]
+            this.selectedQuery.fields[event.dropIndex] = temp
+            this.updateSmartView()
+        },
+        closePreview() {
+            this.qbePreviewDialogVisible = false
+            this.pagination = { start: 0, limit: 25 }
+        },
+        onQueryFieldRemoved(fieldId) {
+            this.selectedQuery.fields.splice(fieldId, 1)
+            this.updateSmartView()
+        }
+    }
+})
+</script>
+<style lang="scss">
+.full-screen-dialog.p-dialog {
+    max-height: 100%;
+    height: 100vh;
+    width: calc(100vw - #{54px});
+    margin: 0;
+}
+
+.full-screen-dialog.p-dialog .p-dialog-header,
+.full-screen-dialog.p-dialog .p-dialog-content {
+    padding: 0;
+    margin: 0;
+}
+
+.full-screen-dialog.p-dialog .p-dialog-content {
+    flex: 1;
+    overflow: hidden;
+}
+
+.entities-lists {
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+    border-right: 1px solid #ccc;
+    height: 100%;
+}
+
+.detail-view {
+    display: flex;
+    flex-direction: column;
+    flex: 3;
+}
+
+.derived-entities-toggle {
+    height: 25%;
+}
+
+.olap-scroll-panel .p-scrollpanel-content {
+    padding: 0 !important;
+}
+.olap-scroll-panel .p-scrollpanel-bar {
+    background-color: #43749eb6;
+    width: 5px;
+}
```

Comment:
Can those classes be scoped? Why there are some olap scroll related classes?

Reply:
Most classes cannot be scoped or targeted with :deep, but i simplified them.
What i could i moved to the descriptor and what could be connected i did so and put them in a same class.
If this is satisfying, we have finished all changes and PR can be accepted!